### PR TITLE
Use DOMReady (bodyComplete) signal where necessary

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: extensions/amp-geo/0.1/amp-geo.js [hunk 1]
```diff
@@ -303,7 +303,7 @@ export class AmpGeo extends AMP.BaseElement {
 
     // Wait for the body before we figure anything out because we might be
     // prerendered and we know that from body classes
-    return ampdoc.whenBodyAvailable().then(body => {
+    return ampdoc.whenReady().then(body => {
```
Comment (jridgewell) [inline extensions/amp-geo/0.1/amp-geo.js]: From the comment, I think this can actually use `whenBodyAvailable `.

Comment (dvoytenko) [reply extensions/amp-geo/0.1/amp-geo.js]: I looked inside and I see that it searches DOM for additional elements. Paging @jpettitt to clarify. Will be happy to change this or split into two phases.

Comment (dvoytenko) [reply extensions/amp-geo/0.1/amp-geo.js]: Will merge this change and revisit this question with @jpettitt at a later time.

Comment (jpettitt) [reply extensions/amp-geo/0.1/amp-geo.js]: `amp-geo` is doing two things that need the body.

1) it needs to be able to read the classes of the `<body>` element to detect if geo was pre-rendered.,

2) it's will append an `amp-state` with the resolved geo to the body.

The first one should be safe as soon as we see the opening body tag.  The second one could be a problem as the position of the inserted amp-state becomes dependent on how much of the body has been parsed.   We should either defer the amp-state append un til the body completes or insert it at the top of the body.  Deferring it is probably the more reasonable approach.

Comment (dvoytenko) [reply extensions/amp-geo/0.1/amp-geo.js]: John, it currently waits for both with this code:

```
return ampdoc
  .whenReady()
  .then(() => ampdoc.whenBodyAvailable())
  .then(body => {
     ...
  })
```

Do you mean this is ok for both (1) and (2)? Or would you rather separate (1) and (2)?

**Pull request merged**